src/I18n/Time.php

```diff
@@ -588,6 +588,10 @@ protected function _formatObject($date, $format, $locale)
             );
         }
 
+        if (preg_match('/@calendar=(japanese|buddhist|chinese|persian|indian|islamic|hebrew|coptic|ethiopic)/', $locale)) {
+            static::$_formatters[$key]->setCalendar(\IntlDateFormatter::TRADITIONAL);
```

Comment:
Or set calendar higher up in the method so the formatter is initialised correctly.